upstairs/src/lib.rs

```diff
@@ -629,28 +723,41 @@ where
 
         // Verify we are in a valid state at this point.
         match my_state {
-            DsState::New
-            | DsState::Disconnected
-            | DsState::Faulted
-            | DsState::Offline => {} // Okay.
+            DsState::New | DsState::Disconnected | DsState::Faulted => {} // Ok
+
+            DsState::Offline => {
+                /*
+                 * This is only applicable for a downstairs that is
+                 * returning from being disconnected. Mark any in progress
+                 * jobs since the last good flush back to New, as we have
+                 * reconnected to this downstairs and will need to replay
+                 * any work that we were holding that we did not flush.
+                 */
+                ds.re_new(up_coms.client_id);
+            }
+            DsState::OnlineRepair => {
+                /*
+                 * Write more code here, when a downstairs is being repaired
```

Comment:
Does this need an XXX or TODO?

Reply:
Yes, this should be part of my next change, but I'll put a TODO here (and the other places)

Reply:
fixed in 02d694383914e8a615080888c1b1bcf9de1be8a3